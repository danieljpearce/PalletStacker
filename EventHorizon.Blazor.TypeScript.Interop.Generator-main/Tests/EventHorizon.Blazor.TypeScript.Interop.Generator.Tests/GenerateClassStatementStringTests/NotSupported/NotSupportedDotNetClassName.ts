declare module Examples {
    export class IDisposable {
        disposeProperty: IDisposable;
        disposeMethod(arg1: IDisposable): IDisposable;
    }
}