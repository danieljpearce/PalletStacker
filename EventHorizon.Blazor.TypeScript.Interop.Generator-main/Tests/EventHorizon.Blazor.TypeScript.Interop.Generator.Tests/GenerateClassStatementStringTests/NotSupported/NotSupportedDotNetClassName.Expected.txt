/// Generated - Do Not Edit
namespace Examples
{
    using System;
    using System.Collections.Generic;
    using System.Text.Json.Serialization;
    using System.Threading.Tasks;
    using EventHorizon.Blazor.Interop;
    using EventHorizon.Blazor.Interop.Callbacks;
    using EventHorizon.Blazor.Interop.ResultCallbacks;
    using Microsoft.JSInterop;

    
    
    [JsonConverter(typeof(CachedEntityConverter<_IDisposable>))]
    public class _IDisposable : CachedEntityObject
    {
        #region Static Accessors

        #endregion

        #region Static Properties

        #endregion

        #region Static Methods

        #endregion

        #region Accessors

        #endregion

        #region Properties
        private _IDisposable __disposeProperty;
        public _IDisposable disposeProperty
        {
            get
            {
            if(__disposeProperty == null)
            {
                __disposeProperty = EventHorizonBlazorInterop.GetClass<_IDisposable>(
                    this.___guid,
                    "disposeProperty",
                    (entity) =>
                    {
                        return new _IDisposable() { ___guid = entity.___guid };
                    }
                );
            }
            return __disposeProperty;
            }
            set
            {
__disposeProperty = null;
                EventHorizonBlazorInterop.Set(
                    this.___guid,
                    "disposeProperty",
                    value
                );
            }
        }
        #endregion
        
        #region Constructor
        public _IDisposable() : base() { }

        public _IDisposable(
            ICachedEntity entity
        ) : base(entity)
        {
            ___guid = entity.___guid;
        }


        #endregion

        #region Methods
        public _IDisposable disposeMethod(_IDisposable arg1)
        {
            return EventHorizonBlazorInterop.FuncClass<_IDisposable>(
                entity => new _IDisposable() { ___guid = entity.___guid },
                new object[]
                {
                    new string[] { this.___guid, "disposeMethod" }, arg1
                }
            );
        }
        #endregion
    }
}